skyportal/handlers/api/spectrum.py

```diff
@@ -790,3 +797,90 @@ def get(self):
 
         self.verify_and_commit()
         return self.success(data=query.all())
+
+
+class SyntheticPhotometryHandler(BaseHandler):
+    @auth_or_token
+    def post(self, spectrum_id):
+        """
+        ---
+        description: Create synthetic photometry from a spectrum
+        tags:
+          - spectra
+        parameters:
+          - in: path
+            name: spectrum_id
+            required: true
+            schema:
+              type: integer
+          - in: query
+            name: filters
+            schema:
+              type: list
+            required: true
+            description: |
+                List of filters
+        responses:
+          200:
+            content:
+              application/json:
+                schema: SingleSpectrum
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        filters = data.get('filters')
+
+        spectrum = Spectrum.get_if_accessible_by(
+            spectrum_id,
+            self.current_user,
+            raise_if_none=True,
+        )
+
+        spec_dict = recursive_to_dict(spectrum)
+        wav = spec_dict['wavelengths']
+        flux = spec_dict['fluxes']
+        err = spec_dict['errors']
+        print(spectrum.astropy_units)
+        obstime = spec_dict['observed_at']
+
+        try:
+            spec = sncosmo.Spectrum(
+                wav, flux * spectrum.astropy_units, err * spectrum.astropy_units
+            )
+        except TypeError:
+            spec = sncosmo.Spectrum(wav, flux * spectrum.astropy_units)
+
+        data_out = []
+        for filt in filters:
+            try:
+                mag = spec.bandmag(filt, magsys='ab')
+                magerr = 0
+            except ValueError:
+                continue
```

Comment:
This seems kind of reckless...

Reply:
I am adding a log message. What else do you think?

Reply:
I'm not sure. In what cases would this happen? Failure to post one of the photometry points doesn't necessarily mean the whole operation should be stopped... but I also don't want the user to feel like it all worked smoothly and never notice some of the filters were not posted. 
It may be the right thing to return a self.error if any of the filters fail, unless there's a general reason for this to happen often. If there is such a reason, we should be very clear to report that reason to the user.

Reply:
Good point, will raise an error.